tools/modm_tools/elf2uf2.py

```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2021, Andrey Kunitsyn
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# Authors:
+# - 2022, Andrey Kunitsyn
+
+import os
+import struct
+
+from pathlib import Path
+
+verbose = False
+
+# we require 256 (as this is the page size supported by the device)
+LOG2_PAGE_SIZE = 8 # @todo configurable
+PAGE_SIZE = (1 << LOG2_PAGE_SIZE)
+
+# @todo read from modm-device
```

Comment:
Perhaps you can pass this map as an argument like it's [done in the `size.py` tool](https://github.com/modm-io/modm/blob/develop/tools/modm_tools/size.py#L20-L22).
Scons puts the memory map in the environment as `CONFIG_DEVICE_MEMORY` and [calls the tool directly](https://github.com/modm-io/modm/blob/develop/tools/build_script_generator/scons/site_tools/size.py#L31-L36).
Not entirely sure if the formats are compatible though.